### PR TITLE
Multiple distributions

### PR DESCRIPTION
**Pull request by daikitag**
Need check from Jerome before merging

### CODE REVIEW

**Comment by daikitag** [comment]
Failing pytests, as I wrote a new function for obtaining the trait model

File: tstrait/trait_model.py [hunk 1]
```diff
@@ -140,8 +91,110 @@ def sim_effect_size(self, num_causal, allele_freq, rng):
...
+        self.df = df
+    
+    def sim_effect_size(self, num_causal, allele_freq, rng):
+        super().sim_effect_size(num_causal, allele_freq, rng)
+        beta = rng.standard_t(self.df)
+        beta = (beta*np.sqrt(self.trait_var) + self.trait_mean)/num_causal
+        beta *= np.sqrt(pow(2 * allele_freq * (1 - allele_freq), self.alpha))
+        return beta
+
+class TraitModelGamma(TraitModel):
```
Comment (gregorgorjanc) [inline tstrait/trait_model.py:152]: @daikitag make sure you flip some effects to negative in the Gamma model, say using Bernoulli samples.

Comment (daikitag) [reply tstrait/trait_model.py:152]: @gregorgorjanc 
Thank you very much for your feedback. I really appreciate it.

**Comment by daikitag** [comment]
I have done the following:
- Finish writing the new codes to select the distribution
- Modify `__init__.py` file to import the correct functions and classes
- Finish writing unit tests for the trait_model() function
- Delete unit tests and statistical tests for the overall `sim_phenotype()` function. The trait model specification has changed, and we are planning to change the architecture of `sim_phenotype()`, so it would be better to write these tests for the new functions instead.

File: tstrait/simulate_phenotype.py [hunk 2]
```diff
@@ -250,7 +253,10 @@ def sim_genetic_value(self):
                 2 * len(individual_genotype)
             )
             beta_array[i] = self.model.sim_effect_size(
```
Comment (jeromekelleher) [inline tstrait/simulate_phenotype.py:255]: This looks good @daikitag , but I think we can simplify a bit by entirely decoupling the frequency dependence for beta from the trait model like so:
```python
beta = self.model.sim_effect_size(self.num_causal, self.rng)
beta_array[i] = frequency_dependence(allele_frequency[i], self.alpha)
```

So the trait distribution really is just drawing ``num_causal`` values from a given numpy random distribution.
```

**Review by jeromekelleher — COMMENTED** [review]
Looks good @daikitag, but I think we can simplify a bit more by taking the frequency dependence entirely out of the code for drawing betas from the a given distribution.

It looks like you've deleted some files here (stats_tests.py and test_simulate_phenotype.py) which shouldn't be deleted, and also you have some unrelated docs changes. I think you probably need to do a rebase against upstream/main.

Please follow the instructions [here](https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html#rebasing) for this.

If it doesn't go OK the first time, don't worry. I can do it easily if needs be.

File: tstrait/simulate_phenotype.py (hunk 2 above)
Comment (daikitag) [reply tstrait/simulate_phenotype.py]: @jeromekelleher 
Thank you very much for your feedback. I modified the codes to add the frequency dependence in the simulation part, and I also restructured the unit tests and statistical tests.

**Comment by daikitag** [comment]
I think there is need to raise an error message if the site ID in effect size dataframe is not included in the inputted tree sequence data in the phenotype simulation step.
https://github.com/tskit-dev/tstrait/issues/49

**Comment by jeromekelleher** [comment]
It's looks like you've gone ahead with a lot of other changes here as well as the updates for multiple distributions @daikitag, which is making this quite hard to review.

The history is also quite messy, so it's not clear what changes are new and what's incidental. It's really not a good idea to use your forks ``main`` branch for development - please use feature branches as described in the tskit dev docs.

Can you pull out just the updates to ``trait_model.py`` along with extracting frequency dependance into a new branch and PR please?

Do something like
```
cp trait_model.py trait_model.py.backup
git fetch upstream
git checkout upstream/main
git checkout -b additional-models
cp trait_model.py.backup trait_model.py
```

You should be able to update the tests without too much duplication.

Then, we can make another feature branch to develop the separate trait simulations

**Pull request closed without merging**